datafusion/common/src/column.rs

```diff
@@ -112,12 +114,18 @@ impl Column {
 
     /// Serialize column into a quoted flat name string
     pub fn quoted_flat_name(&self) -> String {
-        // TODO: quote identifiers only when special characters present
-        // see: https://github.com/apache/arrow-datafusion/issues/5523
         match &self.relation {
             Some(r) => {
-                format!("{}.{}", r.to_quoted_string(), quote_identifier(&self.name))
+                let regexp = Regex::new(r"^[a-zA-Z][_a-zA-Z0-9]*$").unwrap();
+                let column_name = if regexp.is_match(&self.name) {
+                    self.name.clone()
+                } else {
+                    quote_identifier(&self.name)
+                };
+
+                format!("{}.{}", r.to_quoted_string(), column_name)
```

Comment:
probably worth a follow on pr to handle similar logic in `to_quoted_string()` for `TableReference`

Reply:
It is a good idea. I will open a new ticket for it.